WooCommerce/Classes/ViewRelated/ReusableViews/LabeledTextViewTableViewCell.swift

```diff
@@ -0,0 +1,88 @@
+import UIKit
+import Yosemite
+/// A table view cell that contains a label and a text view.
+///
+final class LabeledTextViewTableViewCell: UITableViewCell {
+    struct ViewModel {
+        var text: String? = nil
+        var productStatus: ProductStatus
+        var placeholder: String? = nil
+        var textViewMinimumHeight: CGFloat? = nil
+        var isScrollEnabled: Bool = true
+        var keyboardType: UIKeyboardType = .default
+        var onNameChange: ((_ text: String) -> Void)? = nil
+        var onTextDidBeginEditing: (() -> Void)? = nil
+        var style: Style = .headline
+
+    }
+    @IBOutlet weak var productLabelHolder: UIView!
+    @IBOutlet weak var productStatusLabel: UILabel!
+    @IBOutlet var productTextField: EnhancedTextView!
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        configureLabelStyle()
+        configureBackground()
+    }
+
+    func configure(with viewModel: ViewModel) {
+        productTextField.text = viewModel.text
+        productTextField.placeholder = viewModel.placeholder
+        productStatusLabel.text = viewModel.productStatus.description
+        productTextField.isScrollEnabled = viewModel.isScrollEnabled
+        productTextField.onTextChange = viewModel.onNameChange
+        productTextField.onTextDidBeginEditing = viewModel.onTextDidBeginEditing
+        productTextField.keyboardType = viewModel.keyboardType
+        configureProductStatusLabel(productStatus: viewModel.productStatus)
+    }
+}
+
+private extension LabeledTextViewTableViewCell {
+
+    func configureBackground() {
+        productTextField.backgroundColor = .systemColor(.secondarySystemGroupedBackground)
+    }
+
+
+    func configureLabelStyle() {
+        productStatusLabel.font = UIFont.preferredFont(forTextStyle: .body)
+        productStatusLabel.textAlignment = .center
+        productStatusLabel.textColor = UIColor.black
+        productLabelHolder.backgroundColor = .gray(.shade5)
+        productLabelHolder.layer.cornerRadius = CGFloat(4.0)
+    }
+
+    func configureProductStatusLabel(productStatus: ProductStatus) {
+            if productStatus == ProductStatus.draft {
+                let statusLabel = NSLocalizedString(productStatus.description, comment: "Display label for the product's draft status")
+
+                productStatusLabel.isHidden = false
+                productStatusLabel.text? = statusLabel
+            } else {
+                productLabelHolder.isHidden = true
```

Comment:
The previous block in the if/else is setting `isHidden` on `productStatusLabel`, but this one is changing `productLabelHolder`. I think both should change the holder?

Right now, if I product is a draft and you change the status the badge disappears, but if a product is published, and you revert to draft, the badge won't show up until you close the product view and open it again.